pkg/util/testutil/timeout.go

```diff
@@ -13,6 +13,18 @@ import (
 // Condition is evaluated in a goroutine to avoid tests hanging if a system
 // is deadlocked.
 func AssertTrueBeforeTimeout(t *testing.T, frequency, timeout time.Duration, condition func() bool) {
+	internalTrueBeforeTimeout(t, false, frequency, timeout, condition)
+}
+
+// RequireTrueBeforeTimeout is the same as AssertTrueBeforeTimeout, but it calls
+// t.failNow() if the condition function times out.
+func RequireTrueBeforeTimeout(t *testing.T, frequency, timeout time.Duration, condition func() bool) {
+	if !internalTrueBeforeTimeout(t, true, frequency, timeout, condition) {
```

Comment:
since you're giving the information to `internalTrueBeforeTimeout` whether it should use `require` or `assert`, you could use that info in `internalTrueBeforeTimeout`, for example with:

```go
fail := assert.Fail 
if require {
  fail = require.Fail
}
```

and then use `fail` instead of `assert.Fail`, and remove this logic here in `RequireTrueBeforeTimeout`

Reply:
This is not (easily) possible, since the two functions have a different signature (`assert.Fail` returns bool, while `require.Fail` doesn't return anything, so there is no type (AFAICS) that the `fail` variable could have that would allow assigning any of these.